src/main/java/edu/umich/ctools/sectionsUtilityTool/SectionsUtilityToolServlet.java

```diff
@@ -643,10 +672,48 @@ private boolean isAllowedApiRequest(HttpServletRequest request) {
 		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_CROSSLIST)) && session.getAttribute(LAUNCH_TYPE).equals("lti")){
 			isAllowedRequest = isCrosslistAllowed(request, session, url);
 		}
+		
+		if( url.matches(appExtPropertiesFile.getProperty(CANVAS_API_ADD_USER)) && request.getSession().getAttribute(LAUNCH_TYPE).equals("lti")){
```

Comment:
Maybe wrap these two in a block that check that it is an LTI call and then check the crosslist / add_user value?  It's a bit easier then to see the special aspects of LTI processing.